charmcraft.yaml

```diff
@@ -38,12 +48,36 @@ config:
         `nts://ntp.example.com`
       type: string
       default: ""
+    nts-certificates:
+      descriptions: >-
+        A comma-separated list of juju secret IDs for enabling NTS with. 
+        Each juju secret ID must be a juju secret URI (i.e. start with `secret:`). 
+        The secret must include two fields: the `cert` field, which contains the PEM-encoded TLS certificate, 
+        and the `key` field, which contains an unencrypted PEM-encoded TLS key. 
+        You can create and grant access to the secret using the `juju add-secret` and `juju grant-secret` commands.
+        For instance:
+        `juju add-secret my-tls-cert cert#file=/path/to/fullchain.pem key#file=/path/to/key.pem`
+        `juju grant-secret my-tls-cert chrony`
+      type: string
+      default: ""
```

Comment:
Juju secrets are stored in a `dict[str, str]` format, meaning that if a user needs to store structured data as a secret value, it must be encoded in YAML or JSON. I have some concerns about this approach, particularly regarding the need for users to manually edit sensitive files such as `fullchain.pem` and `key.pem` using a text editor. Additionally, this format could complicate the use of the Juju terraform provider for provisioning secrets.

Reply:
I'm not a big fan of using comma separated string as configuration, but I guess we don't have so many alternatives since secrets aren't discoverable by the charm and must be explicitly configured.